services/fee-estimator/shared/dynamicFees.js

```diff
@@ -99,19 +106,33 @@ const calculateFeePerByte = async block => {
 		block.transactions,
 		async tx => {
 			// Calculate minFee from JSON parsed transaction
-			const schema = await getAllCommandsParamsSchemas();
-			const paramsSchema = schema.commands
-				.find(s => s.moduleID === tx.moduleID && s.commandID === tx.commandID);
-			const parsedTxParams = parseInputBySchema(tx.params, paramsSchema);
+			const metadata = await requestConnector('getSystemMetadata');
+			const filteredModule = metadata.modules.find(module => module.name === tx.module);
+			const filteredCommand = filteredModule.commands.find(s => s.name === tx.command);
+
+			const { params } = tx;
+			const decodedParams = codec.decode(filteredCommand.params, Buffer.from(params, 'hex'));
+			const parsedTxParams = parseInputBySchema(decodedParams, filteredCommand.params);
+
+			const schema = await requestConnector('getSchema');
 			const parsedTx = parseInputBySchema(tx, schema.transaction);
-			const minFee = await getTxnMinFee({
-				...parsedTx,
-				params: parsedTxParams,
-			});
+			const parsedTxWithParams = { ...parsedTx, params: parsedTxParams };
+
+			const txBuffer = codec.encode(
+				schema.transaction,
+				{ ...parsedTx, params: Buffer.from(params, 'hex') },
+			);
+
+			// TODO: Remove when transaction ID is available from SDK
+			tx.id = hash(txBuffer).toString('hex');
```

Comment:
Please assign the `tx.id` and `tx.size` within the connector. Also, the `tx.params` should also be decoded within the `blockchain-connector`. Please ensure that none of this logic is present outside of the  `blockchain-connector`.

Reply:
Can be address in the separate issue.